corgi/core/models.py

```diff
@@ -324,16 +324,19 @@ def save_product_taxonomy(self):
 
         return None
 
-    def disassociate_with_product(self, product_model_name: str, product_pk: str) -> None:
-        """Remove the product references from all components associated with this build."""
+    def disassociate_with_product(
```

Comment:
This looks fine. More an FYI. I have changed the way this works in PR712 so inspect clear and reset the relations for each component related to the software build. It will make this change redundant, but if you need it now please merge as is.